pytorch_pfn_extras/training/extensions/accumulate/_accumulate_base.py

```diff
@@ -0,0 +1,69 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Tuple
+
+import torch.distributed
+from pytorch_pfn_extras import reporting
+from pytorch_pfn_extras.training import extension
+from pytorch_pfn_extras.training._manager_protocol import (
+    ExtensionsManagerProtocol,
+)
+from pytorch_pfn_extras.training.extensions.accumulate._summary import (
+    SummaryBase,
+)
+from pytorch_pfn_extras.training.trigger import TriggerLike, get_trigger
+
+
+class AccumulateBase(ABC, extension.Extension):
+    priority = extension.PRIORITY_EDITOR
+
+    def __init__(
+        self,
+        conversion_key_pair: Tuple[str, str],
+        trigger: TriggerLike = (1, "epoch"),
+        distributed: bool = False,
+    ) -> None:
+        self._conversion_key_pair = conversion_key_pair
+        self._trigger = get_trigger(trigger=trigger)
+        self._distributed = distributed
+        if not torch.distributed.is_initialized() and self._distributed:  # type: ignore[no-untyped-call]
+            raise RuntimeError("PyTorch distributed module is not initialized.")
+
+        self._init_summary()
+
+    def __call__(self, manager: ExtensionsManagerProtocol) -> None:
+        observation = manager.observation
+        src_key, dst_key = self._conversion_key_pair
+        self._summary.add(observation[src_key])
+
+        if self._trigger(manager=manager):
+            if self._distributed:
+                summary = self._all_reduce_summaries()
```

Comment:
The `distributed` flag is intended to allow values stored for newly reported keys by this accumulate to be the ones that are all_reduced across processes. If want to control the timing of all_reduce with an independent trigger, it can be difficult for users to understand the meaning of the stored values.

In the current implementation, the number of times all_reduce is performed is minimized within the range satisfying the specifications. Therefore, if want to control the number of all_reduce executions, it can be achieved by adjusting the trigger argument.